src/github.com/getlantern/flashlight/proxied/proxied.go

```diff
@@ -300,22 +365,13 @@ func httpClient(rootCA string, proxyAddrFN eventual.Getter, persistent bool) (*h
 		}
 	}
 
-	if proxyAddrFN != nil {
-		log.Debug("Waiting for proxy server to come online")
-		proxyAddr, ok := proxyAddrFN(60 * time.Second)
+	tr.Proxy = func(req *http.Request) (*url.URL, error) {
+		proxyAddr, ok := getProxyAddr()
 		if !ok {
-			// Instead of finishing here we just log the error and continue, the client
-			// we are going to create will surely fail when used and return errors,
-			// those errors should be handled by the code that depends on such client.
-			log.Errorf("Proxy never came online")
+			return nil, ErrChainedProxyUnavailable
```

Comment:
Note - we no longer fall back to a direct connection.  As far as I can tell, we never used this, and the clients weren't expecting this to go direct.  Direct connections are also a fingerprinting/blocking threat, so I took it out, which simplified the code too.